src/Views/InstalledView.vala

```diff
@@ -67,6 +67,8 @@ public class AppCenter.Views.InstalledView : View {
 
         yield refresh_mutex.lock ();
 
+        refresh_cancellable.reset ();
+
```

Comment:
We cancel the cancellable at the start of this method to try and get the previous operation to finish earlier so we can acquire the lock. So we need to reset the cancellable before running anything that checks it, or else we cancel the operation before it even starts.